core/schema/host.go

```diff
@@ -72,6 +76,47 @@ func (s *hostSchema) envVariableSecret(ctx *router.Context, parent *core.HostVar
 	return core.NewSecretFromHostEnv(parent.Name), nil
 }
 
+type setSecretFileArgs struct {
+	Name string
+	Path string
+}
+
+func (s *hostSchema) setSecretFile(ctx *router.Context, parent any, args setSecretFileArgs) (*core.Secret, error) {
+	if s.host.DisableRW {
+		return nil, core.ErrHostRWDisabled
+	}
+
+	var absPath string
+	var err error
+	if filepath.IsAbs(args.Path) {
+		absPath = args.Path
+	} else {
+		absPath = filepath.Join(s.host.Workdir, args.Path)
+	}
+
+	absPath, err = filepath.EvalSymlinks(absPath)
+	if err != nil {
+		return nil, fmt.Errorf("eval symlinks: %w", err)
+	}
+
+	file, err := os.Open(absPath)
```

Comment:
Looks like this is missing a `defer file.Close()`, but you could instead just use [`os.ReadFile`](https://pkg.go.dev/os#ReadFile)

Reply:
Ooooh yes, replaced it with the `os.Readfile` 🙏